itest/rust/src/object_tests/property_template_test.rs

```diff
@@ -0,0 +1,95 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+//! Testing that GDScript and rust produces the same property info for properties exported to Godot.
+
+// We're using some weird formatting just for simplicity's sake.
+#![allow(non_snake_case)]
+
+use std::collections::HashMap;
+
+use crate::framework::itest;
+use godot::engine::global::PropertyUsageFlags;
+use godot::prelude::*;
+use godot::sys::GdextBuild;
+
+use crate::framework::TestContext;
+
+use crate::register_tests::gen_ffi::PropertyTemplateRust;
+
+#[itest]
+fn property_template_test(ctx: &TestContext) {
+    let rust_properties = Gd::<PropertyTemplateRust>::new_default();
+    let gdscript_properties = ctx.property_template.clone();
+
+    // Accumulate errors so we can catch all of them in one go.
+    let mut errors: Vec<String> = Vec::new();
+    let mut properties: HashMap<String, Dictionary> = HashMap::new();
+
+    for property in rust_properties.get_property_list().iter_shared() {
+        let name = property
+            .get("name")
+            .unwrap()
+            .to::<GodotString>()
+            .to_string();
+
+        // The format of array-properties in Godot 4.2 changed. This doesn't seem to cause issues if we
+        // compile against 4.1 and provide the property in the format 4.1 expects but run it with Godot 4.2.
+        // However this test checks that our output matches that of Godot, and so would fail in this
+        // circumstance. So here i'm just ignoring array properties when we compile for 4.1 but run in 4.2.
```

Comment:
Nitpick:
```suggestion
        // The format of array-properties in Godot 4.2 changed. This doesn't seem to cause issues if we
        // compile against 4.1 and provide the property in the format 4.1 expects, but run it with Godot 4.2.
        // However this test checks that our output matches that of Godot, and so would fail in this
        // circumstance. So for now, just ignore array properties when we compile for 4.1 but run in 4.2.
```